tools/clang/unittests/HLSL/PixTestUtils.cpp

```diff
@@ -274,6 +274,57 @@ PassOutput RunAnnotationPasses(dxc::DxcDllSupport &dllSupport, IDxcBlob *dxil,
           Tokenize(outputText.c_str(), "\n")};
 }
 
+class InstructionOffsetSeekerImpl : public InstructionOffsetSeeker {
+
+  std::map<std::wstring, DWORD> m_labelToInstructionOffset;
+
+public:
+  InstructionOffsetSeekerImpl(DebuggerInterfaces &debuggerInterfaces) {
+    DWORD SourceFileOrdinal = 0;
+    CComBSTR fileName;
+    CComBSTR fileContent;
+    while (SUCCEEDED(debuggerInterfaces.compilationInfo->GetSourceFile(
+        SourceFileOrdinal, &fileName, &fileContent))) {
+      auto lines = strtok(std::wstring(fileContent), L"\n");
+      for (size_t line = 0; line < lines.size(); ++line) {
+        const wchar_t DebugLocLabel[] = L"debug-loc(";
+        auto DebugLocPos = lines[line].find(DebugLocLabel);
+        if (DebugLocPos != std::wstring::npos) {
+          auto StartLabelPos = DebugLocPos + (_countof(DebugLocLabel) - 1);
+          auto CloseDebugLocPos = lines[line].find(L")", StartLabelPos);
+          VERIFY_ARE_NOT_EQUAL(CloseDebugLocPos, std::string::npos);
+          auto Label = lines[line].substr(StartLabelPos,
+                                          CloseDebugLocPos - StartLabelPos);
+          CComPtr<IDxcPixDxilInstructionOffsets> InstructionOffsets;
+          VERIFY_SUCCEEDED(debuggerInterfaces.debugInfo
+                               ->InstructionOffsetsFromSourceLocation(
+                                   fileName, line + 1, 0, &InstructionOffsets));
+          auto InstructionOffsetCount = InstructionOffsets->GetCount();
+          VERIFY_IS_TRUE(InstructionOffsetCount > 0);
+          // No duplicate labels:
+          VERIFY_IS_TRUE(m_labelToInstructionOffset.find(Label) ==
+                         m_labelToInstructionOffset.end());
+          // Just the last offset is sufficient:
+          m_labelToInstructionOffset[Label] =
+              InstructionOffsets->GetOffsetByIndex(InstructionOffsetCount - 1);
```

Comment:
Isn't source location to instruction offset a many-to-many mapping, or at least a one to many?  Shouldn't this capture all the offsets associated with the source location?  Why not just capture the `IDxcPixDxilInstructionOffsets`?

If capturing only one instruction offset, doesn't that create a problem if you wanted to step and stop at the first offset associated with the location (in this case you're only storing the last)?  For instance, you could have an unrolled loop, and you want to stop on the first instruction associated with that location at each iteration.

Reply:
Should such a need arise, it seems an easy thing to add. A speculative test-support feature doesn't seem worth it for now.